clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsBuilderTest.java

```diff
@@ -66,25 +65,38 @@ public String toString() {
         }
     }
 
+    private static Stream<Arguments> allArguments(Predicate<CompressionType> accept) {
+        List<Arguments> values = new ArrayList<>();
+        for (int bufferOffset : Arrays.asList(0, 15))
+            for (CompressionType compressionType : CompressionType.values())
+                if (accept.test(compressionType))
+                    values.add(Arguments.of(new Args(bufferOffset, compressionType)));
+        return values.stream();
+    }
+
     private static class MemoryRecordsBuilderArgumentsProvider implements ArgumentsProvider {
         @Override
         public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
-            List<Arguments> values = new ArrayList<>();
-            for (int bufferOffset : Arrays.asList(0, 15))
-                for (CompressionType compressionType : CompressionType.values())
-                    values.add(Arguments.of(new Args(bufferOffset, compressionType)));
-            return values.stream();
+            return allArguments(type -> true);
+        }
+    }
+
+    /**
+     *   CompressionType.ZSTD only support when magic >= RecordBatch.MAGIC_VALUE_V2
+     */
```

Comment:
@chia7712   Is it the comment that  you want to add?

Reply:
How about "ZSTD is not supported for MAGIC_VALUE_V0 and MAGIC_VALUE_V1"?